apps/system/js/global_overlay_window_manager.js

```diff
@@ -0,0 +1,111 @@
+/* global Service */
+
+'use strict';
+
+(function(exports) {
+  var GlobalOverlayWindowManager = function() {};
+  GlobalOverlayWindowManager.prototype = {
+    DEBUG: true,
+    TRACE: false,
+    CLASS_NAME: 'GlobalOverlayWindowManager',
+
+    // We only allow one GlobalOverlayWindow instance at a time.
+    // Allowing multiple global overlay windows would lead us to weird UIs.
+    _instance: null,
+
+    publish: function(eventName, detail) {
+      this.debug('publishing: ', eventName);
+      var evt = new CustomEvent(eventName, { detail: detail });
+      window.dispatchEvent(evt);
+    },
+
+    debug: function() {
+      if (this.DEBUG) {
+        console.log('[' + this.CLASS_NAME + ']' +
+          '[' + Service.currentTime() + '] ' +
+          Array.slice(arguments).concat());
+        if (this.TRACE) {
+          console.trace();
+        }
+      }
+    },
+
+    screen: document.getElementById('screen'),
+
+    start: function owm_start() {
+      window.addEventListener('globaloverlaycreated', this);
+      window.addEventListener('globaloverlayrequestclose', this);
+      window.addEventListener('globaloverlayrequestopen', this);
+      window.addEventListener('globaloverlayclosed', this);
+      window.addEventListener('globaloverlayterminated', this);
+
+      window.addEventListener('system-resize', this);
+    },
+
+    stop: function owm_stop() {
+      this._instance = null;
+
+      window.removeEventListener('globaloverlaycreated', this);
+      window.removeEventListener('globaloverlayrequestclose', this);
+      window.removeEventListener('globaloverlayrequestopen', this);
+      window.removeEventListener('globaloverlayclosed', this);
+      window.removeEventListener('globaloverlayterminated', this);
+
+      window.removeEventListener('system-resize', this);
+    },
+
+    handleEvent: function owm_handleEvent(evt) {
+      this.debug('handling ' + evt.type);
+
+      var overlay = evt.detail;
+
+      if (!this._instance && evt.type != 'globaloverlaycreated') {
+        return;
+      }
+
+      switch (evt.type) {
+        case 'globaloverlaycreated':
+          if (this._instance) {
+            console.error('We can only have one global overlay at a time');
+            overlay.kill();
+            return;
+          }
+          this._instance = overlay;
+          break;
+
+        case 'globaloverlayrequestclose':
+          if (this._instance == overlay) {
+            this._instance.close();
+          }
+          break;
+
+        case 'globaloverlayrequestopen':
+          if (this._instance != overlay) {
+            return;
+          }
+          this._instance.ready(function() {
+            this._instance.setVisible(true);
+            this._instance.open();
+          }.bind(this));
+          break;
+
+        case 'globaloverlayterminated':
+          if (this._instance == overlay) {
+            this._instance = null;
+          }
+          break;
+
+        case 'system-resize':
```

Comment:
I guessed that we need to be resized to wrap the hole screen. Otherwise, we might have an overlay window overlaying only part of the screen. Am I right?